In [176]:
import pandas as pd
import os

In [177]:
directory = os.getcwd()
gender_submission = "gender_submission.csv"
test = "test.csv"
train = "train.csv"
gender_submission_path = os.path.join(directory, gender_submission)
test_path = os.path.join(directory, test)
train_path = os.path.join(directory, train)
gender_submission_df = pd.read_csv(gender_submission_path)
test_df = pd.read_csv(test_path)
train_df = pd.read_csv(train_path)

In [178]:
df_people = pd.concat([train_df, test_df], join="outer")
df_people = df_people.set_index("PassengerId")

In [179]:
gender_submission_df = gender_submission_df.set_index("PassengerId")
df_people.update(gender_submission_df)

In [195]:
df_people.head()

Survived  Pclass  \
PassengerId                     
1                 0.0       3   
2                 1.0       1   
3                 1.0       3   
4                 1.0       1   
5                 0.0       3   

                                                          Name     Sex   Age  \
PassengerId                                                                    
1                                      Braund, Mr. Owen Harris    male  22.0   
2            Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.0   
3                                       Heikkinen, Miss. Laina  female  26.0   
4                 Futrelle, Mrs. Jacques Heath (Lily May Peel)  female  35.0   
5                                     Allen, Mr. William Henry    male  35.0   

             SibSp  Parch            Ticket     Fare Cabin Embarked  
PassengerId                                                          
1                1      0         A/5 21171   7.2500   NaN        S  
2                1      0          PC 17599  71.2833   C85        C  
3                0      0  STON/O2. 3101282   7.9250   NaN        S  
4                1      0            113803  53.1000  C123        S  
5                0      0            373450   8.0500   NaN        S

## 1. ¿Cuántas personas iban en el titanic?

In [190]:
df_people["Name"].count()

1309

### Hay 1309 nombres distintos. Eso significa que había 1309 personas en el titanic

## 2. ¿Cuántos hombres y mujeres sobrevivieron?

In [280]:
df_people.groupby(["Survived", "Sex"])["Name"].count()

Survived  Sex   
0.0       female     81
          male      734
1.0       female    385
          male      109
Name: Name, dtype: int64

### Suponiendo que 0 significa que no sobrevivieron (False) y 1 que sí (True), hubo 385 mujeres y 109 hombres que sobrevivieron

## 3. ¿Cuál fue el top 10 de edad que más sobrevieron y el top 10 de edad que no lo lograron?

Dos métodos para obtener esta información. La primera es utilizando las propiedades de pandas y la segunda es mediante agrupaciones.

In [295]:
df_people.loc[df_people['Survived'] == 1]['Age'].value_counts().head(10)

22.0    21
24.0    20
18.0    16
36.0    16
30.0    16
27.0    15
35.0    14
29.0    13
45.0    12
19.0    12
Name: Age, dtype: int64

In [266]:
df_people_groupby_survived_ages = df_people.groupby(["Survived", "Age"])["Name"].count().reset_index(name="count")

In [279]:
df_not_survivers = df_people_groupby_Survived_Ages.loc[df_people_groupby_Survived_Ages["Survived"]==0.0]
df_not_survivers.sort_values(by="count", ascending=False).head(10)

Survived   Age  count
26       0.0  21.0     33
33       0.0  25.0     27
31       0.0  24.0     27
40       0.0  30.0     24
37       0.0  28.0     23
21       0.0  18.0     23
27       0.0  22.0     22
34       0.0  26.0     20
23       0.0  19.0     17
39       0.0  29.0     17

### El top 10 de sobrevivientes es: 22, 24, 18, 36, 30,27, 35, 29, 45 y 19

### El top 10 de no sobrevivientes es: 21, 25, 24, 30, 28, 18, 22, 26, 19 y 29

### Cabe destacar 12 personas cuya edad es menor de 1 con decimales. Se asume que esto es debido a que son personas de apenas unos meses. Por si acaso, se analizan a continuación:

In [300]:
babies = df_people.loc[df_people['Age'] < 1]

In [303]:
babies['Survived'].value_counts()

1.0    9
0.0    3
Name: Survived, dtype: int64